src/main/java/cart/service/cart/CartService.java

```diff
@@ -57,30 +66,31 @@ public void updateQuantity(final Member member, final Long cartItemId, final Car
         validateOwner(cart, cartItem);
 
         if (request.getQuantity() == EMPTY) {
-            cart.removeItem(cartItem);
             cartRepository.removeCartItem(cartItem);
             return;
         }
 
-        cart.changeQuantity(cartItem.getId(), request.getQuantity());
-        cartItem.changeQuantity(request.getQuantity());
-        cartRepository.updateCartItemQuantity(cartItem);
+        cartRepository.updateCartItemQuantity(cartItem, request.getQuantity());
+    }
+
+    private void validateOwner(final Cart cart, final CartItem cartItem) {
+        if (!cartRepository.hasCartItem(cart, cartItem)) {
+            throw new MemberNotOwnerException();
+        }
     }
 
     @Transactional
     public void remove(final Member member, final Long cartItemId) {
         CartItem cartItem = cartRepository.findCartItemById(cartItemId);
```

Comment:
cartItemId로 Cart를 찾아오는 로직을 만든다면 해당 클래스에서 로직전개가 조금더 깔끔해질거 같아요 ㅎ
```suggestion
        Cart cart = cartRepository.findCartItemById(cartItemId);
        // 멤버 비교 등등
```

Reply:
`CartItemId`가 있는 `cartId`라서 그렇군요!

레포지토리를 살짝 수정해서 이렇게 한다면 Member와 Cart의 주인이 같은지 확인할 때에도 더욱 명확하게 떨어지겠네요.